provisionersdk/scripts/bootstrap_darwin.sh

```diff
@@ -31,4 +31,12 @@ fi
 
 export CODER_AGENT_AUTH="${AUTH_TYPE}"
 export CODER_AGENT_URL="${ACCESS_URL}"
-exec ./$BINARY_NAME agent
+
+output=$(./${BINARY_NAME} --version | head -n1)
```

Comment:
I can't recall now, but I worry if `${}` can be interpreted as a terraform variable here. This is good practice but I think we should avoid it in the bootstrap scripts.

Reply:
It's a good call-out, but `BINARY_NAME` is not replaced by the provider, it seems:
https://github.com/coder/terraform-provider-coder/blob/7815596401d6e69aebb4ceefe1e84369cb63c4ac/provider/agent.go#L345-L376

IMHO I think we should use a different template replacement syntax than Bash's variable expansion, to make it very clear that these are replaced by a script and not accepted into the script as env vars.